dockerclient/client.go

```diff
@@ -1079,27 +1079,29 @@ func (e *ClientExecutor) Archive(fromFS bool, src, dst string, allowDownload boo
 		klog.V(5).Infof("Archiving %s %s -> %s from context archive", e.ContextArchive, src, dst)
 		return archiveFromFile(e.ContextArchive, src, dst, excludes, check)
 	}
-	// if the source is an archive, extract it under the destination directory
-	srcFullPath := filepath.Join(e.Directory, src)
-	if allowDownload && isArchivePath(srcFullPath) {
-		f, err := os.Open(srcFullPath)
-		if err != nil {
-			return nil, nil, fmt.Errorf("error opening %q: %v", srcFullPath, err)
-		}
-		klog.V(5).Infof("Extracting %s -> %s from local archive", srcFullPath, dst)
-		transform := func(h *tar.Header, r io.Reader) (data []byte, update bool, skip bool, err error) {
-			h.Name = filepath.Join(dst, h.Name)
-			if h.Typeflag == tar.TypeLink {
-				h.Linkname = filepath.Join(dst, h.Linkname)
+	if false {
+		// if the source is an archive, extract it under the destination directory
+		srcFullPath := filepath.Join(e.Directory, src)
+		if allowDownload && isArchivePath(srcFullPath) {
+			f, err := os.Open(srcFullPath)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error opening %q: %v", srcFullPath, err)
 			}
-			return nil, false, false, nil
-		}
-		filtered, err := transformArchive(f, true, transform)
-		if err != nil {
-			f.Close()
-			return nil, nil, fmt.Errorf("error transforming archive %q: %v", srcFullPath, err)
+			klog.V(5).Infof("Extracting %s -> %s from local archive", srcFullPath, dst)
+			transform := func(h *tar.Header, r io.Reader) (data []byte, update bool, skip bool, err error) {
+				h.Name = filepath.Join(dst, h.Name)
+				if h.Typeflag == tar.TypeLink {
+					h.Linkname = filepath.Join(dst, h.Linkname)
+				}
+				return nil, false, false, nil
+			}
+			filtered, err := transformArchive(f, true, transform)
+			if err != nil {
+				f.Close()
+				return nil, nil, fmt.Errorf("error transforming archive %q: %v", srcFullPath, err)
```

Comment:
Potential stutter?

Reply:
Not sure, but replacing "error" with "while".

Reply:
Removed this, too.